Generate API docs (#1884)
nginxinc/nginx-gateway-fabric: site/content/how-to/monitoring/tracing.md

This guide explains how to enable tracing on HTTPRoutes using NGINX Gateway Fabric. It uses the OpenTelemetry Collector and Jaeger to process and collect the traces.

> To see all the possible configurations for our resources, pleas refer to the [API reference]({{< relref "reference/api.md" >}}).

Reviewer: ```suggestion
Fo all the possible configuration options for these resources, see to the [API reference]({{< relref "reference/api.md" >}}).
```

I would actually include this down below in the Enable Tracing section after we list the required policies, instead of here.

Reviewer: Also include this in the client settings doc, as well as in a `Further reading` section at the bottom of each of these docs.

Author: That client settings doc hasn't been merged yet, right?

Reviewer: Oh you're right

Reviewer 2: FYI: client settings doc has been merged

Author: Do we need to mention it twice? In a further reading section too?

Reviewer: the only reason I thought it could be beneficial is if someone skimmed over the first link and went through the guide, and then at the end wanted to check out the full spec

> - `ObservabilityPolicy`: This resource is a [Direct PolicyAttachment](https://gateway-api.sigs.k8s.io/reference/policy-attachment/) that targets HTTPRoutes or GRPCRoutes. It is created by the [application developer](https://gateway-api.sigs.k8s.io/concepts/roles-and-personas/) and enables tracing for a specific route or routes. It requires the `NginxProxy` resource to exist in order to complete the tracing configuration.

Reviewer 3: shall we also update https://deploy-preview-1884--nginx-gateway-fabric.netlify.app/nginx-gateway-fabric/how-to/configuration/control-plane-configuration/ similarly?

Author: Are you saying to link the API reference in that doc too?

Reviewer 3: I think we can give the link and replace the spec section with a link to here https://deploy-preview-1884--nginx-gateway-fabric.netlify.app/nginx-gateway-fabric/reference/api/#gateway.nginx.org/v1alpha1.NginxGatewaySpec

Reviewer 3: issue created https://github.com/nginxinc/nginx-gateway-fabric/issues/2119

## Further Reading

- [Custom policies]({{< relref "overview/custom-policies.md" >}}): learn about how NGINX Gateway Fabric custom policies work.

> - [API reference]({{< relref "reference/api.md" >}}): learn about the API reference for NGINX Gateway Fabric.

Reviewer: ```suggestion
- [API reference]({{< relref "reference/api.md" >}}): all configuration fields for the policies mentioned in this guide
```
